feat: support rich text from body

EndBug opened the pull request:
Closes #24, closes #51 

This PR makes the action use Martian to parse the issue body and add it to Notion. If the issue has been edited the existing entry will be edited as expected.

marissamarym reviewed (commented):
I think this will just add rich text to the table- for this issue, we want to add the issue body as blocks on the page associated with a table row
<img width="385" alt="Screen Shot 2022-02-21 at 2 18 26 PM" src="https://user-images.githubusercontent.com/1459660/155032602-8bd7392e-4bb3-4245-833e-632531afd8f2.png">



<img width="973" alt="Screen Shot 2022-02-21 at 2 18 38 PM" src="https://user-images.githubusercontent.com/1459660/155032579-67771fe8-929e-44a0-a2c4-20ae129feb2c.png">
.

EndBug commented:
Alright, now it correctly updates the blocks inside the page as intended:
<img width="732" alt="image" src="https://user-images.githubusercontent.com/26386270/155188282-297fb2f3-472b-4b9c-910a-fcd21a00ffa0.png">

The blocks are added if not present, removed if more than needed, and updated when possible

The "Body" field has to be removed from the Notion template, since it's not used anymore.

marissamarym commented on src/sync.ts:
```diff
@@ -138,7 +147,6 @@ function getPropertiesFromIssue(issue: Issue): CustomValueMap {
     milestone,
     created_at,
     updated_at,
-    body,
```
Can we keep the body field as well? People might want to use it as a filter in the table

EndBug replied:
Sure, do we want to display everything or only something like the first lines?
Because if we display everything then the page will have to copies of the same body

marissamarym commented on src/action.ts:
```diff
@@ -94,6 +112,36 @@ async function handleIssueEdited(options: IssueEditedOptions) {
+
+  await Promise.all(
+    bodyBlocks.slice(0, overlap).map((block, index) =>
+      notion.client.blocks.update({
```
Is there anything in the api
that lets us batch these updates? And can we save calls by not updating blocks that haven't changed?

EndBug replied:
Unfortunately, it seems like the [Notion API](https://developers.notion.com/reference/patch-page) doesn't have a way to bulk send these updates: Ones the children are added there's no way to overwrite them in one go, you have to use the delete and update endpoints for each of them. 
An alternative could be deleting the page and creating a new one, which would avoid the calls to every single block, but I guess it would break links and the order on the table (?)

marissamarym commented on src/action.ts:
```diff
@@ -94,6 +112,36 @@ async function handleIssueEdited(options: IssueEditedOptions) {
+    });
+  } else if (bodyBlocks.length < existingList.length) {
+    await Promise.all(
+      existingList.slice(overlap).map(block => notion.client.blocks.delete({block_id: block.id}))
```
Similar to the comment above, does anything in the api let us delete all blocks in a page in one call?

EndBug commented:
Alright, the action now updates both the table field and the children

I haven't added a check regarding [the request limits for property values](https://developers.notion.com/reference/request-limits#limits-for-property-values) since that part is handled by `martian`, which already has an open issue about that: https://github.com/instantish/martian/issues/6

Unfortunately, it doesn't seem that the Notion API (which is still in beta) has yet added a way to batch block updates and deletions, so right now we have to stick to sending a request for each one. At the moment we're only adding one paragraph block per issue, so the request count is relatively low.
We also can't detect the blocks that have changed: we would have to send a GET request for each one of them in order to compare them with the updated version, which kind of defeats the purpose...

The only alternative that comes to mind is to delete and re-create the database entry from scratch every time, but I guess it would mess the links and position in the db, and as of now (with only one child block) it would probably be less efficient.

Let me know what you think 👍🏻
